### DISCUSSION TITLE
Heater not working

### DISCUSSION THREAD

**Discussion by Princerivo** [DanWBR/dwsim#557] [General]
Hi,
    For some reason when I am attaching a heater for my model, the program is not responding at all. I am using the last version of the program. Any thoughts on that is appreciated.

Regards,
 M

**Comment by DanWBR** [comment]
what is your operating system and dwsim version?

**Comment by Princerivo** [comment]
Hi Daniel,
     Thanks for your kind response. I use Windows 11 and DWSIM version 8.6.5.

Regards,
M

**Reply by DanWBR** [reply]
can you share the flowsheet file?

**Comment by Princerivo** [comment]
Please see attached...

[Heater_Case.zip](https://github.com/DanWBR/dwsim/files/13453249/Heater_Case.zip)

**Reply by DanWBR** [reply]
You had a loop around the heater and needed to add a recycle block, check the attached file.
[Heater_Case2.zip](https://github.com/DanWBR/dwsim/files/13453263/Heater_Case2.zip)

**Comment by Princerivo** [comment]
Many thanks for the quick response, very much appreciated. But could I ask is it a must for any loop to have the recycle block?

Regards,
M.

**Reply by DanWBR** [reply]
yes, it is. DWSIM doesn't detect these tear streams automatically.

**Reply by Princerivo** [reply]
Thanks alot...

Regards,
M.
